contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -1491,6 +1492,8 @@ pltsql_strpos_non_determinstic(text *src_text, text *substr_text, Oid collid, in
 		src_ulen = icu_to_uchar(&src_uchar, VARDATA_ANY(src_text), src_len_utf8);
 		substr_ulen = icu_to_uchar(&substr_uchar, VARDATA_ANY(substr_text), substr_len_utf8);
 
+		is_substr_starts_with_surrogate = U16_IS_SURROGATE(substr_uchar[0]);
```

Comment:
What if search str is empty?

Reply:
empty string never reaches pltsql_strpos_non_determinstic.
Even if it did we will error out before we reach here. Confirmed by setting values using GDB.

Reply:
We also have test cases for null inputs/empty string inputs in charindex, replace and patindex.